### PR TITLE
Surface diff fixup

### PR DESCRIPTION
**Pull request by ChrisBNEU**
### Motivation or Problem
currently RMG cannot use the "diffmodels" tool to compare surface+gas chemkin files. This means that we can't use a surface model in the CI tests, which is a big problem. This PR is a step towards fixing that. 

### Description of Changes
This PR combines fixes for several issues:
-  interpreting surface input files
-  combining chemkin gas+surface output files
-  comparing surface species/reactions. 

### Testing
Currently have tested by creating a diff.html file for a combined gas+surface model as shown in this [minimal example](https://github.com/ReactionMechanismGenerator/RMG-Py/files/9588247/min_example_for_pr.zip) 

### Reviewer Tips
Until we add in the a surface mechanism to our functional/CI tests, this will not be used regularly, so try it with a surface mechanism and make sure the output is captured as we expect.

### CODE REVIEW

File: rmgpy/kinetics/surface.pyx [hunk 1]
```diff
@@ -422,6 +438,22 @@ cdef class StickingCoefficientBEP(KineticsModel):
...
+        Returns ``True`` if rates of reaction at temperatures 500,1000,1500,2000 K
+        and 1 and 10 bar are within +/ .5 for log(k), in other words, within a factor of 3.
+        """
+        cdef double T
+
+        if not isinstance(other_kinetics, StickingCoefficientBEP):
+            return False
+
+        for T in [500, 1000, 1500, 2000]:
+            if abs(self.get_sticking_coefficient(T) - other_kinetics.get_sticking_coefficient(T)) > 1e-7:
```
Comment (rwest) [inline rmgpy/kinetics/surface.pyx:452]: again, is it 0.5 or 1e-7?

**Review by rwest — COMMENTED** [review]
Not finished review yet but battery going flat

File: rmgpy/kinetics/surface.pyx [hunk 2]
```diff
@@ -220,6 +220,22 @@ cdef class StickingCoefficient(KineticsModel):
...
+        Returns ``True`` if rates of reaction at temperatures 500,1000,1500,2000 K
+        and 1 and 10 bar are within +/ .5 for log(k), in other words, within a factor of 3.
+        """
+        cdef double T
+
+        if not isinstance(other_kinetics, StickingCoefficient):
+            return False
+
+        for T in [500, 1000, 1500, 2000]:
+            if abs(self.get_sticking_coefficient(T) - other_kinetics.get_sticking_coefficient(T)) > 1e-7:
```
[the remark this answers is not in the document]
Comment (ChrisBNEU) [reply rmgpy/kinetics/surface.pyx:234]: Took a look back at the code,, and now I think I changed my mind. I don't see a good reason why we wouldn't treat sticking coefficient rxns like all the other reaction types. I can change it and push.

[the remark this answers is not in the document]
Comment (rwest) [reply rmgpy/kinetics/surface.pyx]: I'm not sure what the best transform is for a sticking coefficient (which is always between 0 and 1) to be "similar". 
To my gut, 0.00001 feels "similar" to 0.00003  (a factor of 3x) but 0.33 does not feel very "similar" to 0.99 (a factor of 3x).

**Comment by ChrisBNEU** [comment]
It looks like this PR it is passing everything except the RMS_liquidSurface_ch4o2cat regression test (which I think is addressed in @hwpang's pr). Should I change the CI file to skip it so I can ensure the remaining actions are successful or should we just review and merge?

**Comment by hwpang** [comment]
@ChrisBNEU If you rebase with the newest main, the CI for the liquid surface reactor is turned off by https://github.com/ReactionMechanismGenerator/RMG-Py/pull/2473!

**Comment by JacksonBurns** [comment]
> It looks like this PR it is passing everything except the RMS_liquidSurface_ch4o2cat regression test (which I think is addressed in @hwpang's pr). Should I change the CI file to skip it so I can ensure the remaining actions are successful or should we just review and merge?

@rwest removed that regression test in #2473, please rebase and then all should be fine.

**Review by rwest — APPROVED** [review]

**Pull request merged**